### PR TITLE
feat(gui-v2): Grid operations

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

- [x] Row delete
- [x] Multiple rows select and delete
- [ ] Add a new row
- [ ] Column delete
- [ ] Set column as primary ( todo: need to highlight the cell )
- [ ] Add debounce to avoid frequent API calls
- Keyboard navigations
    - [ ] Tab / Shift + Tab to navigate through cells
    - [ ] Arrow key for navigating through cell
    - [ ] Enter key for making editable
    - [ ] Delete to clear the cell value
- Context menu
    - [ ] For row delete
    - [ ] Multiple row delete
    - [ ] Clear cell
    - [ ] Insert new row 

re #2911  #2962

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Comment by dstala** [comment]
- [x] Row delete
- [x] Multiple rows select and delete
- [ ] Add a new row
- [ ] Column delete
- [ ] Set column as primary ( todo: need to highlight the cell )
- [ ] Add debounce to avoid frequent API calls
- Keyboard navigations
    - [ ] Tab / Shift + Tab to navigate through cells
    - [ ] Arrow key for navigating through cell
    - [ ] Enter key for making editable
    - [ ] Delete to clear the cell value
- Context menu
    - [ ] For row delete
    - [ ] Multiple row delete
    - [ ] Clear cell
    - [ ] Insert new row

**Comment by dstala** [comment]
@pranavxc : open items summarised here

- [ ] Primary value: highlight cell
- [ ] Increase de-bounce timeout; skip de-bounce for decimal
- [ ] Enter key to mark cell as editable
- [ ] Support for click & type on cell
- [ ] Right click menu persists during the following set of events
  -  right-click, do not select any option
  - click on edit column; edit menu appears & right-click menu doesn't disappear
  
Screenshot 2022-08-03 at 9 43 01 AM

File: packages/nc-gui-v2/components/smartsheet-header/Cell.vue [hunk 1]
```diff
@@ -1,79 +1,19 @@
 <script setup lang="ts">
+import { Modal } from 'ant-design-vue'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet-header/Cell.vue:2]: unused import

File: packages/nc-gui-v2/components/smartsheet-header/Menu.vue [hunk 2]
```diff
@@ -1,10 +1,54 @@
...
+const meta = inject(MetaInj)
+
+const { $api } = useNuxtApp()
+const { t } = useI18n()
+const toast = useToast()
+const { getMeta } = useMetas()
+
+const deleteColumn = () =>
+  Modal.confirm({
+    title: h('div', { innerHTML: `Do you want to delete <span class="font-weight-bold">'${column?.title}'</span> column ?` }),
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet-header/Menu.vue:28]: instead of `innerHTML` use the 3rg argument for `h` and pass the string as a slot.

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

**Pull request merged**